### ISSUE TITLE
Loading channels fails every other time when using personal access token

### ISSUE THREAD

**Issue by tareko** [nextcloud/integration_mattermost#26]
Mattermost Version: 8.1.0 , running on kubernetes
Nextcloud version: 27.0.2
Integration version: 1.0.4

Behaviour:
When using personal access tokens, every second attempt to share is met with an 'unauthorized' error:

```Client error: `GET https://x/api/v4/users/x/channels` resulted in a `401 Unauthorized` response:{"id":"api.context.session_expired.app_error","message":"Invalid or expired session, please login again.","detailed_erro (truncated...)```

The channels list subsequently doesn't load. If I repeat this, usually the second time it will load the channel list. Occasionally, it doesn't load them all and leaves blanks. And sometimes (presumably because there are requests happening in the background), it will error out the next time too.

[Screencast from 2023-09-14 13-37-43.webm](https://github.com/nextcloud/integration_mattermost/assets/1128840/d15cdd99-5903-4b9b-9ad7-8f8bd4396c08)

One thing to note here is that my mattermost server is running via kubernetes with 2 replicas. Very early on, I had an issue in which authentication sessions wouldn't 'stick' as it went back and forth between both servers, however that hasn't been an issue since very early and certainly isn't an issue now anywhere but here.

Of note, this error _does not_ happen when using username / password login, so I am able to use everything normally. This is only when using PATs, which I assume is a minority of users anyhow, making this a bit of a corner case.

I'm happy to troubleshoot more when the server is not as busy and I can reduce to one replica to confirm that this is indeed the root cause. However, not sure what to do about it if so.

Thank you!

**Comment by stepcellwolf** [comment]
Yes, I'm having the same issue. I tried with perosnal token and with OAuth 2.0 Applications still the same. Any workaround?
